posthog/hogql/transforms/property_types.py

```diff
@@ -180,9 +180,11 @@ def _add_property_notice(
         else:
             materialized_column = self._get_materialized_column("events", property_name, "properties")
 
-        message = f"{property_type.capitalize()} property '{property_name}' is of type '{field_type}'"
+        message = f"{property_type.capitalize()} property '{property_name}' is of type '{field_type}'. "
         if materialized_column:
-            message = "⚡️" + message
+            message += "This property is materialized ⚡️."
+        else:
+            message += "This property is not materialized 🐢."
```

Comment:
I think the thought behind this is great, but I suspect this is one of those messages that doesn't make sense to anyone but us, and turns off the "slightly technical" user/makes posthog more confusing.

If there's a way of only showing this in /debug I think that'd be nicer? Or maybe only show this when `user.is_staff` or is inpersonated?

Reply:
What if instead we call these "optimized properties"? The idea has always been to give clever users enough nudges that they're able to figure out why something fails, shows different data, or is slow. Hopefully this would push them to prefer materialized properties, or even come to us directly with "my query is slow, and it says these properties are slow... can we do something?"

Perhaps this is wishful thinking 🤔 ... happy to also just show this level of detail on the `/debug` page.

Reply:
I don't think there is a way to prefer materialized properties, because if you need property with one piece of data… that's just the property you need. The notice is useful but IMO should be on only for staff users.